src/main/scala/io/iohk/ethereum/blockchain/sync/regular/RegularSync.scala

```diff
@@ -39,7 +40,8 @@ class RegularSync(
     syncConfig: SyncConfig,
     ommersPool: ActorRef,
     pendingTransactionsManager: ActorRef,
-    scheduler: Scheduler
+    scheduler: Scheduler,
+    node: BlockchainConfigBuilder
```

Comment:
`node` is a bit surprising in that context. We can probably just call it `configBuilder`

Reply:
Valid remark. My reasoning was that there might be more traits tacked on later (eg. `node: BlockchainConfigBuilder with SyncConfig`.) Maybe this is only strictly needed for mutability. As for the cake pattern the impact will only become apparent later when these traits become abstact instead of responsible for building them. So the dependencies move from the 'cake traits' into these constraints in the constructor arguments.

Does that make sense? Any thoughts?

Reply:
I got the same reaction, and made exactly the same comment on a previous commit entry (removed now).

My gut feeling is that "node" should be called according to its function, and not its state, if i understand correctly the situtation: you expect the config building feature to be provided by a node instance.

Reply:
OK I'm outnumbered.